src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/topic/LuceneKeyphraseExtractor.scala

```diff
@@ -0,0 +1,140 @@
+package com.microsoft.partnercatalyst.fortis.spark.transforms.topic
+
+import org.apache.commons.collections4.trie.PatriciaTrie
+import org.apache.lucene.analysis.Analyzer
+import org.apache.lucene.analysis.ar.ArabicAnalyzer
+import org.apache.lucene.analysis.de.GermanAnalyzer
+import org.apache.lucene.analysis.en.EnglishAnalyzer
+import org.apache.lucene.analysis.es.SpanishAnalyzer
+import org.apache.lucene.analysis.fr.FrenchAnalyzer
+import org.apache.lucene.analysis.hi.HindiAnalyzer
+import org.apache.lucene.analysis.pt.PortugueseAnalyzer
+import org.apache.lucene.analysis.ru.RussianAnalyzer
+import org.apache.lucene.analysis.standard.StandardAnalyzer
+import org.apache.lucene.analysis.tokenattributes.{CharTermAttribute, OffsetAttribute}
+
+import scala.collection.mutable
+
+class LuceneKeyphraseExtractor(language: String, keyphrases: Set[String], maxKeywords: Int = Int.MaxValue) extends KeyphraseExtractor {
+
+  private val analyzer: Analyzer = LuceneKeyphraseExtractor.analyzersByLanguage.getOrElse(language, new StandardAnalyzer())
+  private val keyphraseTries = LuceneKeyphraseExtractor.createTries(analyzer, keyphrases)
+
+  override def extractKeyphrases(text: String): Set[Keyphrase] = {
+    val tokenStream = analyzer.tokenStream(null, text)
+    try {
+      val collectors = keyphraseTries.map(kv=>{
+        KeyphraseCollector(kv._1, kv._2)
```

Comment:
A collector for different lengths of phrases is created to make this implementation easier to read. For example, the watchlist item `fumigación de cultivos ilicitos` will tokenize as something like `fumiga cultivo ilicit` (stems) which will require token windows of size 3 for matching.